bin/spiced/src/lib.rs

```diff
@@ -65,73 +65,60 @@ pub async fn run(args: Args) -> Result<()> {
     let mut df = runtime::datafusion::DataFusion::new();
 
     for ds in &app.datasets {
-        match &ds.source {
-            Some(source) => {
-                let auth_name = match &ds.auth {
-                    Some(auth_name) => auth_name,
-                    None => source,
-                };
-                let data_source: Box<dyn DataSource> = match source.as_str() {
-                    "spice.ai" => {
-                        let spice_auth = auth.get(auth_name);
-                        Box::new(
-                            datasource::flight::Flight::new(
-                                spice_auth,
-                                "https://flight.spiceai.io".to_string(),
-                            )
-                            .await
-                            .context(
-                                UnableToInitializeDataSourceSnafu {
-                                    data_source: source.clone(),
-                                },
-                            )?,
-                        )
-                    }
-                    "dremio" => {
-                        let dremio_auth = auth.get(auth_name);
-                        Box::new(
-                            datasource::flight::Flight::new(
-                                dremio_auth,
-                                "http://dremio-4mimamg7rdeve.eastus.cloudapp.azure.com:32010"
-                                    .to_string(),
-                            )
-                            .await
-                            .context(
-                                UnableToInitializeDataSourceSnafu {
-                                    data_source: source.clone(),
-                                },
-                            )?,
-                        )
-                    }
-                    "debug" => Box::new(datasource::debug::DebugSource {
-                        sleep_duration: Duration::from_secs(1),
-                    }),
-                    _ => UnknownDataSourceSnafu {
-                        data_source: source.clone(),
-                    }
-                    .fail()?,
-                };
-
-                let data_source = Box::leak(data_source);
-
-                df.attach(
-                    &ds.name,
-                    data_source,
-                    databackend::DataBackendType::default(),
+        let mut from_parts: Vec<&str> = ds.from.split('/').collect();
+        let source = from_parts[0];
+        from_parts.remove(0);
+        let dataset_path = from_parts.join(".");
```

Comment:
The idea is that just the string before the first slash is used to select the data source - everything after will be left as is to be interpreted by the data source

Reply:
Yes, this is just temporary alignment of what we have to the dataset spec. Not an overhaul of the OSS internals to the end state.

Reply:
Actually, upon further thought I agree with you that we should start doing this now. I'll make this change too.